### ISSUE TITLE
obtain the distance

### ISSUE THREAD

**Issue by rojinsafavi** [lmcinnes/umap#345]
Is there a way to extract the measured metric? (ex the euclidian distance used for the embedding)

**Comment by lmcinnes** [comment]
Do you mean the distances in the original high dimensional space, or the distances in the low dimensional space? The first of those can be done, but if that's all you want then I would recommend looking at [pynndescent](https://www.github.com/lmcinnes/pynndescent) which is an offshoot project for just handling the fast nearest neighbour distance computation. The distances in the low dimensional space are compyed on the fly in the course of the computation and never really stored anywhere. I think the best option there would be either sklearn's ``pairwise_distances`` or, for large data, use the ``KDTree`` from sklearn and only query the nearby points.

**Comment by rojinsafavi** [comment]
Thanks for the quick response.
So  I am specifically interested in the **correlation** metric that is used in **umap.UMAP** function ( and I assume the correlation distance is applied to the original data). Do you suggest looking into the pynndescent for that?

**Comment by lmcinnes** [comment]
If you just want to compute with that distance then you can:
```python
from umap.distances import correlation
```
and then use that with other tools, such as:
```python
from sklearn.metric import pairwise_distances

dist_mat = pairwise_distances(data, metric=correlation)
```
Which would give you an all pairs distance matrix between your data points using the correlation distance.

In general pynndescent will be useful if you want to get distances to nearest neighbors as efficiently as possible -- which is what UMAP does for the high dimensional distance computations.
